src/mercury.js

```diff
@@ -62,6 +63,19 @@ const Mercury = {
       contentType,
     });
 
+    if (extend) {
+      const extendedTypes = Reflect.ownKeys(extend);
```

Comment:
One minor request here. Can we extract this somewhere, since it seems like the logic is duplicated b/w `mercury.js` and `RootExtractor`?